report/metrics.go

```diff
@@ -0,0 +1,268 @@
+package report
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"time"
+
+	"github.com/mndrix/ps"
+)
+
+// Metrics is a string->metric map.
+type Metrics map[string]Metric
```

Comment:
In fact, we should break out the changes to ps.Map and ps.List (marshalling etc) into our own structs, and then reuse them.